analysis/study_definition.py

```diff
@@ -88,81 +81,438 @@
+        },
         return_expectations={
             "rate": "universal",
-            "category": {"ratios": {"rural": 0.1, "urban": 0.9}},
+            "category": {"ratios": {"PC": 0.05, "PN": 0.05, "PS": 0.05, "": 0.85, }, },
         },
     ),
- 
-    # https://codelists.opensafely.org/codelist/opensafely/asthma-inhaler-salbutamol-medication/2020-04-15/
-    recent_salbutamol_count=patients.with_these_medications(
-        salbutamol_codes,
-        between=["2018-02-01", "2020-02-01"],
-        returning="number_of_matches_in_period",
+
+    # simple care home flag
+    care_home=patients.categorised_as(
+        {
+            1: """care_home_type""",
+            0: "DEFAULT",
+        },
         return_expectations={
-            "incidence": 0.6,
-            "int": {"distribution": "normal", "mean": 8, "stddev": 2},
+            "rate": "universal",
+            "category": {"ratios": {1: 0.15, 0: 0.85, }, },
+        },
+    ),
+
+    ###############################################################################
+    # COVID VACCINATION
+    ###############################################################################
+    # any COVID vaccination (first dose)
+    covid_vacc_date=patients.with_tpp_vaccination_record(
+        target_disease_matches="SARS-2 CORONAVIRUS",
+        on_or_after="2020-12-01",  # check all december to date
+        find_first_match_in_period=True,
+        returning="date",
+        date_format="YYYY-MM-DD",
+        return_expectations={
+            "date": {
+                "earliest": "2020-12-08",  # first vaccine administered on the 8/12
+                "latest": "2021-01-31",
+            }
+        },
+    ),
+    # SECOND DOSE COVID VACCINATION
+    covid_vacc_second_dose_date=patients.with_tpp_vaccination_record(
+        target_disease_matches="SARS-2 CORONAVIRUS",
+        on_or_after="covid_vacc_date + 19 days",
+        find_first_match_in_period=True,
+        returning="date",
+        date_format="YYYY-MM-DD",
+        return_expectations={
+            "date": {
+                "earliest": "2020-12-29",  # first reported second dose administered on the 29/12
+                "latest": latest_date,
+            }
+        },
+    ),
+    # COVID VACCINATION - Pfizer BioNTech
+    covid_vacc_pfizer_date=patients.with_tpp_vaccination_record(
+        product_name_matches="COVID-19 mRNA Vac BNT162b2 30mcg/0.3ml conc for susp for inj multidose vials (Pfizer-BioNTech)",
+        on_or_after="2020-12-01",  # check all december to date
+        find_first_match_in_period=True,
+        returning="date",
+        date_format="YYYY-MM-DD",
+        return_expectations={
+            "date": {
+                "earliest": "2020-12-08",  # first vaccine administered on the 8/12
+                "latest": latest_date,
+            }
+        },
+    ),
+    # COVID VACCINATION - Oxford AZ
+    covid_vacc_oxford_date=patients.with_tpp_vaccination_record(
```

Comment:
just a note on these vaccine-type fields - at present they pick up first dose _of specified type_ , so need to be analysed appropriately e.g. all those with an oxford vaccine record did not necessarily have it as their first dose.

Reply:
ah good to know

Reply:
could be split out into az-first, az-second, etc if that would be more convenient and avoid any confusion

Reply:
thanks - i have added this in